# Physics 21, Spring 2020
## Assignment 3: LIGO Inspirals

In [53]:
from astropy import constants as c
from matplotlib import pyplot as pl
from astropy import units as u
import numpy as np

# Question 1

In [45]:
c.M_sun

<<class 'astropy.constants.iau2015.IAU2015'> name='Solar mass' value=1.988409870698051e+30 uncertainty=4.468805426856864e+25 unit='kg' reference='IAU 2015 Resolution B 3 + CODATA 2018'>

In [42]:
c.m_n

<<class 'astropy.constants.codata2018.CODATA2018'> name='Neutron mass' value=1.67492749804e-27 uncertainty=9.5e-37 unit='kg' reference='CODATA 2018'>

In [43]:
c.m_p

<<class 'astropy.constants.codata2018.CODATA2018'> name='Proton mass' value=1.67262192369e-27 uncertainty=5.1e-37 unit='kg' reference='CODATA 2018'>

In [44]:
1.6 * c.M_sun / ((c.m_p + c.m_p) / 2)

<Quantity 1.90207706e+57>

From the assignment, we know that the mass of a neutron star is about $1-2.2$ $M_\odot$ ~ 1.6 $M_\odot$. As mass of neutron stars is almost entirely just due to protons and neutrons (which are approximately the same mass), we can write the number of nucleons in the neutron star to be $$\frac{1.6 M_\odot}{(m_n + m_p) / 2} \approx 2 \times 10^{57}$$

# Question 2

In [22]:
c.R_sun.to('km')

<Quantity 695700. km>

In [47]:
(2 * c.G * 30 * c.M_sun / c.c ** 2).to('km')

<Quantity 88.59750228 km>

Radius of the Sun = $6.957\times 10^5$ km. 

Radius of neutron stars ~ 10 km.

Radius (Schwarzschild) of black holes (~ 30 $M_\odot$) ~ 90 km.

# Question 3

In [46]:
c.M_sun

<<class 'astropy.constants.iau2015.IAU2015'> name='Solar mass' value=1.988409870698051e+30 uncertainty=4.468805426856864e+25 unit='kg' reference='IAU 2015 Resolution B 3 + CODATA 2018'>

In [40]:
c.G

<<class 'astropy.constants.codata2018.CODATA2018'> name='Gravitational constant' value=6.6743e-11 uncertainty=1.5e-15 unit='m3 / (kg s2)' reference='CODATA 2018'>

In [41]:
c.c

<<class 'astropy.constants.codata2018.CODATA2018'> name='Speed of light in vacuum' value=299792458.0 uncertainty=0.0 unit='m / s' reference='CODATA 2018'>

In [32]:
(c.M_sun * c.G / c.c ** 2).to("km")

<Quantity 1.47662504 km>

In [34]:
(c.M_sun * c.G / c.c ** 3).to("s")

<Quantity 4.92549095e-06 s>

Mass of Sun (in standard mass units) = $2 \times 10^{30}$ kg 

We use the conversion factor $G/c^2$. Mass of Sun (in standard length units) = $1.5$ km

We use the conversion factor $G/c^3$. Mass of Sun (in standard time units) = $5 \times 10^{-6}$ s 

# Question 4

Frequency of gravitational wave emission is twice the orbital frequency, that is $f_{GW} = 2 f_o$. As time period $T = 1/f$, we get that 0.1 s = $T_o = 1/f_o = 2 / f_{GW}$. Finally, this gives us frequency of GWs emitted is 20 Hz.

# Question 5

In [49]:
T = 0.1 * u.s
m1 = 1.4 * c.M_sun
m2 = 1.4 * c.M_sun

By Kepler's Third law, $\frac{a^3}{T^2} = \frac{G (m_1 + m_2)}{4\pi^2}$. So, we can write $a = (\frac{G (m_1 + m_2)T^2}{4\pi^2})^{1/3}$.

In [57]:
a = (((c.G * (m1 + m2) * T ** 2) / (4 * np.pi ** 2)) ** (1/3))
a.to('km')

<Quantity 454.8867738 km>

So, semi-major axis is $a$ ~ 455 km, while the radial separation between the two neutron stars is $2s$ ~ 910 km.

To get velocity, we note that as we have circular orbits ($\dot{r}$ = 0), the gravitational force provides the centripedal force needed. Thus, $\frac{G m_1 m_2}{d^2} = \frac{m_1 v^2}{r}$, or $v^2 = \frac{G m_2 a}{(2a)^2} = \frac{G m}{4a}$.

In [87]:
v = (c.G * m1 / (4 * a)) ** (1/2)
v.to('km/s')

<Quantity 10105.04343048 km / s>

In [83]:
v / c.c

<Quantity 0.0337068>

So, we get a velocity $v \approx 10^{5}$ km/s and $v/c \approx 0.034$. 

# Question 6

In [64]:
ke = 1/2 * m1 * v ** 2
ke.to('erg')

<Quantity 1.42128221e+51 erg>

In [68]:
pe = - c.G * m1 * m2 / (2 * a)
pe.to('erg')

<Quantity -5.68512883e+51 erg>

So, we get that Kinetic Energy $T \approx 1.42 \times 10^{51}$ erg, and Potential Energy $V \approx - 5.69 \times 10^{51}$ erg.

In [70]:
L = m1 * v * a
L

<Quantity 1.27960356e+43 kg m2 / s>

Angular momentum $L \approx 1.28 \times 10^{43}$ kg m$^2$/ s.

# Question 7

In [73]:
a_ICSO = 6 * c.G * m1 / (c.c ** 2)
a_ICSO.to('km')

<Quantity 12.40365032 km>

ICSO is the smallest circular orbit (lowest possible radius $r$) at which a test particle may orbit a massive object, for general relativity. $a_{ICSO} = \frac{6GM}{c^2}$. For our case with $M = 1.4 M_\odot$, we have $a_{ICSO} = 12.4$ km.

# Question 8

Again, from Kepler's Third law, we have $\frac{a^3}{T^2} = \frac{G (m_1 + m_2)}{4\pi^2}$. So, we can write $T = (\frac{4\pi^2 a^3}{G (m_1 + m_2)})^{1/2}$.

In [76]:
T_ICSO = (4 * np.pi ** 2 * a_ICSO ** 3 / (c.G * (m1 + m2))) ** (1/2)
T_ICSO

<Quantity 0.00045027 s>

In [78]:
f_ICSO = 1/T_ICSO
f_ICSO.to('Hz')

<Quantity 2220.90863665 Hz>

In [79]:
fGW_ICSO = 2 * f_ICSO
fGW_ICSO.to('Hz')

<Quantity 4441.8172733 Hz>

We get: time period $T = 0.00045$ s, orbital frequency $f_o = 2220.9$ Hz, and GW frequency $4441.8$ Hz.

# Question 9

In [81]:
(2 * a_ICSO).to('km')

<Quantity 24.80730064 km>

We have semi-major axis $a_{ICSO} = 12.4$ km, and radial separation $2 a_{ICSO} = 24.8$ km.

To get velocity, we note that as we have circular orbits ($\dot{r}$ = 0), the gravitational force provides the centripedal force needed. Thus, $\frac{G m_1 m_2}{d^2} = \frac{m_1 v^2}{r}$, or $v^2 = \frac{G m_2 a}{(2a)^2} = \frac{G m}{4a}$.

In [88]:
v_ICSO = (c.G * m1 / (4 * a_ICSO)) ** (1/2)
v_ICSO.to('km/s')

<Quantity 61194.87923623 km / s>

In [85]:
v_ICSO / c.c

<Quantity 0.20412415>

So, we get a velocity $v_{ICSO} \approx 6.12 \times 10^{5}$ km/s and $v_{ICSO}/c \approx 0.20$. 

# Question 10

In [89]:
ke_ICSO = 1/2 * m1 * v_ICSO ** 2
ke_ICSO.to('erg')

<Quantity 5.21235653e+52 erg>

In [90]:
pe_ICSO = - c.G * m1 * m2 / (2 * a_ICSO)
pe_ICSO.to('erg')

<Quantity -2.08494261e+53 erg>

So, we get that Kinetic Energy $T_{ICSO} \approx 5.21 \times 10^{52}$ erg, and Potential Energy $V_{ICSO} \approx - 2.08 \times 10^{53}$ erg.

In [92]:
L_ICSO = m1 * v_ICSO * a_ICSO
L_ICSO

<Quantity 2.11299535e+42 kg m2 / s>

Angular momentum $L_{ICSO} \approx 2.11 \times 10^{42}$ kg m$^2$/ s

# Question 11 

In [94]:
deltaE = (pe_ICSO + ke_ICSO) - (ke + pe)
deltaE.to('erg')

<Quantity -1.52106849e+53 erg>

As this term is negative, this $|\Delta E| = 1.52 \times 10^{53}$ erg is lost, in the form of gravitational wave radiation. 

# Question 12

In [98]:
flux = np.abs(deltaE) / (4 * np.pi * (1 * u.Mpc) ** 2)
flux.to('J/m^2')

<Quantity 1.27127005 J / m2>

We assume that this energy $|\Delta E|$ is radiated in a spherically symmetrical manner. Then, the flux (energy per unit area) at $D=1$ Mpc is given as: $\frac{|\Delta E|}{4\pi D^2} = 1.27$ J/m$^2$.

# Question 13